src/components/FilterList.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component, PropTypes } from 'react'
+import { ListView, StyleSheet, Text, TextInput, View } from 'react-native'
+
+import { ContentRow } from './ContentRow'
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: '#272822',
+    flex: 1,
+  },
+  searchContainer: {
+    backgroundColor: 'white',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  searchIcon: {
+    paddingLeft: 10,
+    paddingRight: 10,
+  },
+  searchInputContainer: {
+    width: 300,
+  },
+  noFoundResults: {
+    padding: 10,
+  },
+  noFoundResultsText: {
+    color: '#FFF',
+  },
+})
+
+export class FilterList extends Component {
+  static propTypes = {
+    addToPlaylist: PropTypes.func.isRequired,
+    contents: PropTypes.array,
+    title: PropTypes.string,
+  }
+
+  static defaultProps = { contents: [], title: '' }
+
+  constructor(props) {
+    super(props)
+    this.addToPlaylist = this.addToPlaylist.bind(this)
+    this.state = { textFilter: '', filteredContents: [] }
+  }
+
+  componentWillMount() {
+    this.dataSource = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 })
+  }
+
+  addToPlaylist() {
+    this.props.addToPlaylist()
+  }
+
+  render() {
+    const textFilterRegExp = new RegExp(this.state.textFilter, 'i')
+    const isLongEnoughFilter = this.state.textFilter.length >= 1
+    const filteredContents = isLongEnoughFilter ? this.props.contents.filter(
+      ({ fileName }) => fileName.toLowerCase().match(textFilterRegExp)
+    ) : []
```

Comment:
After some deliberation (and seeing that for ternary, this is [the default of eslint's operator-linebreak rule](http://eslint.org/docs/rules/operator-linebreak)), ok for me.
As there are more ternaries in the project, fix it all in another PR, with the addition of the eslint rule?